File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.js
function solution(nums) {
    
    // 숫자 조합
    let primes = 0;
    
    for(let i = 0; i < nums.length; i++){
        
        for(let j = i + 1; j < nums.length; j++){
            
            
            for(let k = j + 1; k < nums.length; k++){
                
                const sum = nums[i] + nums[j] + nums[k];
                
                let isDivided = false;
                // 소수 판별 
                for(let p = 2; p <= Math.floor(Math.sqrt(sum)); p++){
                    
                    if((sum % p) === 0) isDivided = true;
                }
                
                if(!isDivided){
                    primes++
                }
                
                
            }
        }
    }
    
    console.log(primes)
    
    return primes;
}

File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.py
import math

def solution(nums):
    length = len(nums)
    count = 0
    for i in range(length -2):
        for j in range(i+1, length - 1):
            for k in range(j+1, length):
                if isPrime(nums[i]+nums[j]+nums[k]):
                    count += 1
    return count

def isPrime(num):
    for n in range(2, math.isqrt(num) + 1):
        
        if num % n == 0:
            return False
        
    return True